src/plugins/intel_cpu/src/proxy_mem_mgr.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "proxy_mem_mgr.h"
+#include "utils/debug_capabilities.h"
+
+using namespace ov::intel_cpu;
+
+void ProxyMemoryMngr::setManager(MemoryMngrPtr _pMngr) {
+    auto _validated = (_pMngr != m_pMngr);
+    if (_pMngr) {
+        m_pMngr = _pMngr;
+    } else {
+        m_pMngr = m_pOrigMngr;
+    }
+
+    // WA: unconditionally resize to last size
+    if (_validated) {
+        auto res = m_pMngr->resize(m_Size);
+        DEBUG_LOG(this, ", ", m_pMngr, " size ", m_Size, " -> ", m_Size, " resized? ", res, " RawPtr ", getRawPtr());
+    }
```

Comment:
@maxnick I've noticed that this unconditionally resize to last size cannot solve all problem.

https://github.com/openvinotoolkit/openvino/blob/1e69152d9fe75761887693c64d305bcd78d1cf66/src/plugins/intel_cpu/src/cpu_memory.cpp#L136

Reply:
revised and fixed